### ISSUE TITLE
AWS Toolkit hanging Visual Studio waiting for a process start/exit

### ISSUE THREAD

**Issue by davkean** [aws/aws-toolkit-visual-studio#192]
I'm from the Visual Studio performance team and our telemetry is showing that that the AWS toolkit package is hanging Visual Studio invoking and waiting on a process while on the UI thread. Over the past 21 days, it has had the following impact on Visual Studio:

Root|Leaf|Expressed Hits|Total Hits|Duration (75th Percentile)|Delayed Mouse Clicks|Delayed Key Strokes
---|---|---|---|---|---|---
awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.Services.AWSToolkitShellProviderService+<>c__DisplayClass_0+<<ExecuteOnUIThread>b__>d.MoveNext|awstoolkit.lambda.dll!Amazon.AWSToolkit.Lambda.Util.LambdaTesterInstaller.InstallLambdaTester|2,279|9,631|12.07 sec|2,227|98

Expressed delays are situations where the UI hung while within this method and it interrupted/prevented users mouse or keyboard input. Total hits are all the times the UI hung while within this method, regardless of whether input was interrupted. 

The call stack is as follows:

```
awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.Services.AWSToolkitShellProviderService+<>c__DisplayClass_0+<<ExecuteOnUIThread>b__>d.MoveNext
-> awstoolkit.dll!Amazon.AWSToolkit.ToolkitFactory+<>c__DisplayClass_0.<InitializeToolkit>b__ |
--> awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.AWSToolkitPackage+<>c__DisplayClass_0.<InitializeAsync>b__
----> awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.AWSToolkitPackage.InitializeLambdaTesterEventListener
-----> awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.Lambda.LambdaTesterEventListener..ctor
------> awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.Lambda.LambdaTesterEventListener.Initialize
-------> awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.Lambda.LambdaTesterEventListener.OnSolutionOpened
--------> awstoolkitpackage.dll!Amazon.AWSToolkit.VisualStudio.Lambda.LambdaTesterUtilities.EnsureLambdaTesterConfigured
---------> awstoolkit.lambda.dll!Amazon.AWSToolkit.Lambda.LambdaActivator.EnsureLambdaTesterConfigured
----------> awstoolkit.lambda.dll!Amazon.AWSToolkit.Lambda.Util.LambdaTesterInstaller.Install
-----------> awstoolkit.lambda.dll!Amazon.AWSToolkit.Lambda.Util.LambdaTesterInstaller.InstallLambdaTester
------------> system.dll!System.Diagnostics.Process.WaitForExit
```

**To Reproduce**
There is no steps to reproduce, this came entirely from telemetry in real world usage.

**Expected behavior**
This path has async context, please plumb through async all the way up to InstallLambdaTester and switch to a background thread via `await TaskScheduler.Default` before invoking/waiting on the process to exit. This will prevent the UI from hanging while this long running process finishes.

**Computer (please complete the following information):**

- Windows Version: All versions
- Visual Studio Version: All versions
- AWS Toolkit for Visual Studio Version: Unknown

**Additional context**

**Comment by awschristou** [comment]
Hi @davkean thank you for sharing this insight! We will investigate this.

**Comment by awschristou** [comment]
Hi @davkean , we have released a new version of the Toolkit (v1.23.2.0), where we now run the Lambda tester installation routine on a background thread. I am closing the issue, but we will be interested in hearing if this reduces the hotspot that you've reported. 

We'd also be interested in hearing if the extension shows any other hotspots in your data, as separate issues. Thanks again!

**Comment by davkean** [comment]
@awschristou Thanks a lot, great to hear. Will keep you posted if we find any more issues.
